### PR TITLE
manually update dependencies to .NET 5

### LINKED ISSUE #27: Adopt .NET 5.0 builds
We should plug in to the dependency flow and start adopting 5.0 nightly builds.

### PR DESCRIPTION
**Pull request by anurse**
Fixes #27 

I added `Microsoft.NETCore.App` as a Darc-tracked dependency and then did a manual `darc update-dependencies` to get the latest .NET 5 build in. Then I updated all the TFMs. Everything seems to be working and using .NET 5!

Once this is in, I can update darc to bring in new builds at whatever cadence we want. The trade off there is that more frequent updates means we have the latest and greatest, at the cost of more dependency update PRs. The dependency update PRs *should* auto-merge, but we may have to watch out for breaking changes. Also, more updates means more commits that are just "update dependencies" :).

TODO at merge:
* [x] Confirm the packages build `netcoreapp5.0` binaries.
* [x] Decide on update cadence (options: 'none', 'everyDay', 'everyBuild', 'twiceDaily', 'everyWeek') **thoughts?**
* [ ] Update Darc subscription (that's controlled outside the repo so I wanted to get a manual update in first)

### CODE REVIEW

**Comment by Tratcher** [comment]
Also tied in here is locking down the ship schedule. Do we ship in sync with NET 5 previews, or do we need to pin dependencies to the prior preview.

**Comment by anurse** [comment]
As discussed in our meeting today, our plan is to align, so let's move forward with this for now.

**Comment by anurse** [comment]
Thoughts on update cadence? I'm thinking of starting with everyWeek to avoid too much PR/Commit spam. It's totally possible to do one-off updates whenever we want.

File: samples/ReverseProxy.Sample/ReverseProxy.Sample.csproj [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <Project Sdk="Microsoft.NET.Sdk.Web">
 
   <PropertyGroup>
-    <TargetFramework>netcoreapp3.1</TargetFramework>
+    <TargetFramework>netcoreapp5.0</TargetFramework>
```
[the remark this answers is not in the document]
Comment (anurse) [reply samples/ReverseProxy.Sample/ReverseProxy.Sample.csproj:4]: Ah, is that working? I'll give it a shot :)

[the remark this answers is not in the document]
Comment (anurse) [reply samples/ReverseProxy.Sample/ReverseProxy.Sample.csproj:4]: Didn't work on 5.0.100-preview.2.20120.3 (the current SDK version we are using) 😦.

**Pull request merged**